### PR TITLE
Fpr

### PR DESCRIPTION
**Pull request by brentp**
this adds a `strling pull_region` command that's useful for debugging a certain region as it pulls all mates as well, even if they are spread across the genome.

it also reduces false positives by adjusting some parameters, there is still a single read left for hg002 locus: `12	133294574	133294648	G` on GRCh37 that is incorrectly called as an STR containing read.

### CODE REVIEW

**Comment by hdashnow** [comment]
Unfortunately this change causes us to miss most of the true positives. Investigating further.

**Comment by brentp** [comment]
you'll likely have to update the linear model.

File: src/strpkg/collect.nim [hunk 1]
```diff
@@ -86,8 +86,10 @@ proc count(A: Record, bounds:Bounds): int =
     return 0
 
   if read_left < 0: read_left = 0
-
-  return dna[read_left..<read_right].count(bounds.repeat)
+  let S = dna[read_left..<read_right]
```
Comment (brentp) [inline src/strpkg/collect.nim:89]: @hdashnow you can revert this section to test effect

File: src/strpkg/extract.nim [hunk 2]
```diff
@@ -135,7 +135,10 @@ proc add_soft*(cache:var Cache, aln:Record, counts: var Seqs[uint8], opts:Option
                   split: if cig_index == 0: Soft.left else: Soft.right,
                   mapping_quality: aln.mapping_quality,
                   qname: aln.qname
-                  ))
+                  )
+    # require higher repeat percentage for soft-clips
```
Comment (brentp) [inline src/strpkg/extract.nim:139]: or revert this section

File: src/strpkg/extract.nim [hunk 3]
```diff
@@ -176,6 +178,14 @@ proc adjust_by*(A:var tread, B:tread, opts:Options, B_position:uint32): bool =
         A.position = B_position + B.align_length.uint32
 
     # if we have adjusted A based on B, then any split is not informative.
+    if A.split != Soft.none:
+      # the split affects the align-length. we no longer have the original
+      # length, but we can set it to the max, re-check p_repeat and skip if
+      # it's not high enough.
```
Comment (brentp) [inline src/strpkg/extract.nim:184]: or this section

**Comment by hdashnow** [comment]
Yes, linear model will need to be updated once this is settled. But the linear model doesn't impact detection of the locus, only the size estimate. So not causing the issue I described.

File: src/strpkg/extract.nim (hunk 3 above)
Comment (brentp) [reply src/strpkg/extract.nim:184]: we can remove this section after dropping the section in get_repeat that drops soft-clips.

**Comment by hdashnow** [comment]
Now we're able to recover 12/13 known pathogenic loci that we were detecting previously.

**Comment by hdashnow** [comment]
Is this ready to merge do you think @brentp?

File: nim.cfg [hunk 4]
```diff
@@ -1 +1 @@
--d:danger -d:release
+-d:danger -d:release -d:debug
```
Comment (brentp) [inline nim.cfg:1]: you sure you want this as the default?

File: src/strpkg/collect.nim (hunk 1 above)
Comment (brentp) [reply src/strpkg/collect.nim:89]: i already forgot why we did this section. should add a comment.

File: nim.cfg (hunk 4 above)
Comment (hdashnow) [reply nim.cfg:1]: Good catch. Will fix.

**Comment by brentp** [comment]
LGTM

File: src/strpkg/collect.nim (hunk 1 above)
Comment (hdashnow) [reply src/strpkg/collect.nim:89]: I don't remember either...

Comment (hdashnow) [reply src/strpkg/collect.nim:89]: So I think this boils down to:
```if repeats_in_reads < int(read_length * 0.7 / repeat_length):
    repeats_in_reads```

**Pull request merged**